lib/liquid/extensions.rb

```diff
@@ -60,3 +60,9 @@ def to_liquid # :nodoc:
     self
   end
 end
+
+class Proc
+  def to_liquid # :nodoc:
+    self
```

Comment:
Liquid only uses a proc for lazy loading a result, so shouldn't we be calling `.to_liquid` on the result of that call rather than on the Proc itself?  I'm not sure if this is the place to do that though, since the result should be memoized.